### PR TITLE
feat(build): avoid overly strict babel helpers

### LINKED ISSUE #4639: Build fails on Node.js 14.15
## 🐛 Bug description

Build fails on Node.js 14.15

## 🔍 Bug reproduction

```
$ git clone ...
$ yarn
[...]
$ yarn build
yarn run v1.22.10
$ yarn run build:cjs && yarn run build:es && yarn run build:umd
$ rm -rf cjs && BABEL_ENV=cjs babel src --extensions '.js,.ts,.tsx' --out-dir cjs/ --ignore 'src/index.es.ts','**/__tests__','**/__mocks__' --quiet
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in /workspace/instantsearch.js/node_modules/@babel/helper-compilation-targets/package.json
    at throwExportsNotFound (internal/modules/esm/resolve.js:290:9)
    at packageExportsResolve (internal/modules/esm/resolve.js:513:3)
    at resolveExports (internal/modules/cjs/loader.js:432:36)
    at Function.Module._findPath (internal/modules/cjs/loader.js:472:31)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:867:27)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/workspace/instantsearch.js/node_modules/@babel/preset-env/lib/debug.js:8:33)
    at Module._compile (internal/modules/cjs/loader.js:1063:30) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

```
$ node --version
v14.15.4
$ npm --version
7.5.2
```

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

Make the bundlesize slightly smaller by optimising the babel options

**Result**

Since we don't use the cases presented in those plugins as far as I can see (spreading non-iterables, symbols...)

Impact:

- 1.4KB~ for everything loose
- .6KB~ for only the specified transforms

Closes #4639, #4640, #4644

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -88,6 +90,14 @@ module.exports = api => {
         {
           modules,
           targets,
+          loose: true,
+          exclude: [
+            // not needed, we don't have Symbols
+            '@babel/plugin-transform-typeof-symbol',
+            // we are passing an option
+            '@babel/plugin-transform-classes',
+            '@babel/proposal-object-rest-spread',
```
Comment (Haroenv) [inline babel.config.js:99]: do we pass loose: true for all plugins, or specifically for these two?

I'm in favour of passing loose: true generally

File: package.json [hunk 2]
```diff
@@ -143,7 +143,7 @@
   "bundlesize": [
     {
       "path": "./dist/instantsearch.production.min.js",
-      "maxSize": "66 kB"
+      "maxSize": "66.15 kB"
```
Comment (Haroenv) [inline package.json:146]:
```suggestion
      "maxSize": "65 kB"
```

if we go for loose

Comment (Haroenv) [inline package.json:146]:
```suggestion
      "maxSize": "66 kB"
```

If we go for only the specified plugins

**Comment by Haroenv** [comment]
need to investigate which of these Jest was relying on

File: babel.config.js [hunk 3]
```diff
@@ -88,6 +90,14 @@ module.exports = api => {
         {
           modules,
           targets,
+          loose: !isTest,
```
Comment (francoischalifour) [inline babel.config.js:93]: Why do we use a different `loose` value in test environment? That seems error prone.

**Review by francoischalifour — COMMENTED** [review]
Overall, I'm not sure using `loose` (which is risky and unintuitive in some cases) is worth this small bundle size gain.

**Comment by Haroenv** [comment]
do you think we should update babel with the non-loose (bigger) output as in #4644 then @francoischalifour ?

**Comment by francoischalifour** [comment]
Yes—it's `0.15 kB` bigger but with an updated and stable toolchain.

**Pull request closed without merging**